raft/raft.go

```diff
@@ -547,13 +548,23 @@ func (r *raft) bcastHeartbeatWithCtx(ctx []byte) {
 // the commit index changed (in which case the caller should call
 // r.bcastAppend).
 func (r *raft) maybeCommit() bool {
-	// TODO(bmizerany): optimize.. Currently naive
-	mis := make(uint64Slice, 0, len(r.prs))
+	if cap(r.matchBuf) < len(r.prs) {
+		r.matchBuf = make(uint64Slice, len(r.prs))
+	}
+	mis := r.matchBuf[:len(r.prs)]
+	idx := 0
+	sorted := true
 	for _, p := range r.prs {
```

Comment:
it seems that it is low chance that prs happens to be sorted? not sure if this optimization makes sense.

Reply:
In TiKV, we still sort it every time. maybe we can write a benchmark for this.

Reply:
Yeah the change that this is sorted is low - `1/(n!)` if we assume that map iteration is always random. Probably not worth the extra complexity to this function. Removed.